File: Arithmetic_PrimeFactors.pl
#!/usr/bin/perl;
use strict;
use warnings;
use LISTPRIMEFACTORS;

# Useful variables
my ($test_guy, $answer);

# Use the list prime factors module/class
$test_guy = 97;
my $prime_factors = LISTPRIMEFACTORS->new($test_guy);
$answer = $prime_factors->onlyPrimeFactors();
print ("Prime factorising $test_guy gives:\n", join(" X ", @{$answer}), "\n");


print "\n\n";

File: Arithmetic_PrimeFactors.vb
Module Arithmetic_prime_factors

    Sub Main()

        ' Use the list factors Class
        Dim test_guy = 48
        Dim prime_factors As New ListPrimeFactors
        prime_factors._init_(test_guy)
        Dim answer = prime_factors.onlyPrimeFactors()

        Console.WriteLine("Prime factorising " & test_guy & " gives:")
        Console.WriteLine(String.Join(" X ", answer))

    End Sub

End Module
